library/alloc/src/vec/is_zero.rs

```diff
@@ -69,3 +69,36 @@ unsafe impl<T: ?Sized> IsZero for Option<Box<T>> {
         self.is_none()
     }
 }
+
+// `Option<num::NonZeroU32>` and similar have a representation guarantee that
+// they're the same size as the corresponding `u32` type, as well as a guarantee
+// that transmuting between `NonZeroU32` and `Option<num::NonZeroU32>` works.
+// While the documentation officially makes in UB to transmute from `None`,
```

Comment:
Should this be "makes _it_ ub*"?

Reply:
Oh, you're right. Feel free to send a PR if you want ^^'. But this doesn't appear in any public documentation/interface, so it doesn't matter that much.